package-lock.json

```diff
@@ -45,37 +46,34 @@
     },
     "node_modules/@babel/code-frame": {
       "version": "7.12.11",
-      "resolved": "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.12.11.tgz",
       "integrity": "sha512-Zt1yodBx1UcyiePMSkWnU4hPqhwq7hGi2nFL1LeA3EUl+q2LQx16MISgJ0+z7dnmgvP9QtIleuETGOiOH1RcIw==",
       "dev": true,
       "dependencies": {
         "@babel/highlight": "^7.10.4"
       }
     },
     "node_modules/@babel/compat-data": {
-      "version": "7.14.9",
-      "resolved": "https://registry.npmjs.org/@babel/compat-data/-/compat-data-7.14.9.tgz",
-      "integrity": "sha512-p3QjZmMGHDGdpcwEYYWu7i7oJShJvtgMjJeb0W95PPhSm++3lm8YXYOh45Y6iCN9PkZLTZ7CIX5nFrp7pw7TXw==",
+      "version": "7.16.4",
```

Comment:
Did you intend to upgrade all these dependencies? Any particular reason?

Reply:
Nope, not intentional. I just did `npm i` and added the `@fastify/session` lib

Reply:
Not sure if we should keep the package-lock.

Reply:
+1, It should be removed from the repo and added to the `.gitignore` file

Reply:
I have removed the  package-lock

Reply:
@zekth @darkgl0w have you read the big discussion we had about keeping it in #24 ? I still think it is a good idea to have around to know when upstream dependencies have broken the build. To reiterate, I am fully aware that the package lock has no impact on folks installing this package from NPM, but I do think that enforcing a consistent set of dependency versions for the different developers of this project and CI is worth something, just the same way as it would be for an application using this package. Why remove it?

Reply:
> the package lock has no impact on folks installing this package from NPM

So why not having the same behavior as the users of the project? Usually the goto is to freeze your versions for your end usage like applications/servers and such. Regarding lib this has been debated in all the possible ways you can imagine. Just this is the only repo i think which has a `package-lock.json` in the fastify-plugin ecosystem.

Reply:
> So why not having the same behavior as the users of the project?

I think we should try to reveal what will happen to a new user who installs the project I agree -- it should work great out of the box with a fresh set of dependencies locked by someone else. But, I don't think a lockfile prevents that -- I think it actually helps, because of depfu. Without a package lock or depfu, the main branch of CI just starts failing all of a sudden when an upstream dependency breaks something. Or, worse, if there is no activity on the main branch, it will fail the next time but no has has been alerted to that yet. Instead, I'd rather have depfu be doing the bumps to track the most recent versions of everything, and get each bump to have CI run with github action's free minutes so we actually know if the dependency bump works. Then, if something breaks, we get a red pr where we can collaborate to fix it instead of it happening to some unsuspecting contributor trying to do something else completely. It's happened a couple times already with TypeScript on this repo where a new version of TS was released that breaks our build -- wouldn't you rather know about that as soon as it happens instead of having some user of the package waste their time discovering that and debugging it?

Reply:
Well i don't have strong opinion it's more for ecosystem consistency.

Sidenote: #413 issue is caused by Jest which we should avoid. Node-tap has a pretty good TS integration.

Reply:
@sameer-coder just to unblock you, can you not remove the package-lock.json file from this PR, and the package maintainers can discuss that in a separate issue? That way you are unblocked to get this awesome change in!

Reply:
@airhorns > as you suggested we should keep going on this PR and open a separate issue regarding the `package-lock.json` file removal and the use of jest in this repo.

Reply:
@airhorns @darkgl0w I have added the package-lock back and rebased this PR. If everything looks good then let's merge it!